iphone/Maps/UI/Search/Tabs/HistoryTab/SearchHistoryViewController.swift

```diff
@@ -0,0 +1,93 @@
+protocol SearchHistoryViewControllerDelegate: AnyObject {
+  func searchHistoryViewController(_ viewController: SearchHistoryViewController,
+                                   didSelect query: String)
+}
+
+final class SearchHistoryViewController: MWMViewController {
+  private weak var delegate: SearchHistoryViewControllerDelegate?
+  private var lastQueries: [String]
+  private let frameworkHelper: MWMSearchFrameworkHelper
+  private static let clearCellIdentifier = "SearchHistoryViewController_clearCellIdentifier"
+  
+  @IBOutlet private var tableView: UITableView!
+  @IBOutlet private weak var noResultsViewContainer: UIView!
+  
+  init(frameworkHelper: MWMSearchFrameworkHelper, delegate: SearchHistoryViewControllerDelegate?) {
+    self.delegate = delegate
+    self.lastQueries = frameworkHelper.lastSearchQueries()
+    self.frameworkHelper = frameworkHelper
+    super.init(nibName: nil, bundle: nil)
+  }
+  
+  required init?(coder aDecoder: NSCoder) {
+    fatalError("init(coder:) has not been implemented")
+  }
+  
+  override func viewDidLoad() {
+    super.viewDidLoad()
+    
+    if frameworkHelper.isSearchHistoryEmpty() {
+      showNoResultsView()
+    } else {
+      tableView.register(cellClass: SearchHistoryQueryCell.self)
+      let nib = UINib(nibName: "SearchHistoryClearCell", bundle: nil)
+      tableView.register(nib, forCellReuseIdentifier: SearchHistoryViewController.clearCellIdentifier)
+    }
+  }
+  
+  func showNoResultsView() {
+    guard let noResultsView = MWMSearchNoResults.view(with: UIImage(named: "img_search_history"),
+                                                      title: L("search_history_title"),
+                                                      text: L("search_history_text")) else {
+                                                        assertionFailure()
+                                                        return
+    }
+    noResultsViewContainer.addSubview(noResultsView)
+    tableView.isHidden = true
+  }
+  
+  func clearSearchHistory() {
+    frameworkHelper.clearSearchHistory()
+    lastQueries = []
+  }
+}
+
+extension SearchHistoryViewController: UITableViewDataSource {
+  func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
+    return !frameworkHelper.isSearchHistoryEmpty() ? lastQueries.count + 1 : 0
```

Comment:
If I understand correctly, table view is hidden if there is no search history. So you can safely return `lastQueries.count + 1 `